### ISSUE TITLE
add language to filtering please

### ISSUE THREAD

**Issue by tugrulz** [DocNow/twarc#266]
there is location parameter, but not language :/

**Comment by mirsamantajbakhsh** [comment]
I've found [--lang LANG] option. As it is mentioned in README:

Twitter attempts to code the language of a tweet, and you can limit your search to a particular language if you want:

`twarc search '#blacklivesmatter' --lang fr > tweets.json`

**Comment by edsu** [comment]
Yes, good find, and I just double checked that it still works:

```
twarc search obama --lang fr | jq .full_text
"RT @JeanMessiha: Pendant la présidentielle, le candidat #Macron avait été OFFICIELLEMENT soutenu par #Obama et par #Merkel (qui l’a reçu en…"
"RT @EnzoMarano_RN: \" #Trump est plus populaire que Obama et Reagan au même moment de leur mandat \"\n\n@BFMTV obligé d'assumer encore la réali…"
"RT @greenpeacefr: En sept ans de campagne #SaveTheArctic, de nombreues étapes ont été franchies. \n\n1⃣Nous avons empêché Shell de forer en A…"
"RT @GWGoldnadel: Quelqu’un pourrait il m’expliquer gentiment pour quelles puissantes raisons le soutien d’Obama à Macron aura soulevé moins…"
"RT @GWGoldnadel: Quelqu’un pourrait il m’expliquer gentiment pour quelles puissantes raisons le soutien d’Obama à Macron aura soulevé moins…"
"RT @GWGoldnadel: Quelqu’un pourrait il m’expliquer gentiment pour quelles puissantes raisons le soutien d’Obama à Macron aura soulevé moins…"
"RT @GWGoldnadel: Quelqu’un pourrait il m’expliquer gentiment pour quelles puissantes raisons le soutien d’Obama à Macron aura soulevé moins…"
"https://t.co/yv4h9sR6iN : 4651d6f7-f082-425a-b532-0cfb5c938e42"
```

**Comment by tugrulz** [comment]
I mean twarc filter not twarc search. Filter uses Streaming API. Search uses Search API.

**Comment by edsu** [comment]
Whoops yes, reopened!

**Comment by mirsamantajbakhsh** [comment]
Correct. I've checked your mentioned error. When --lang is used with filter API (stream api) the log says not acceptable. Here is a piece of the log file when running `twarc filter --lang fa`:

```
2019-05-22 18:14:25,589 INFO posting ('https://stream.twitter.com/1.1/statuses/filter.json', {'stall_warning': True, 'include_ext_alt_text': True}) {'headers': {'accept-encoding': 'deflate, gzip'}, 'stream': True}
2019-05-22 18:14:25,843 ERROR caught http error 406 Client Error: Not Acceptable for url: https://stream.twitter.com/1.1/statuses/filter.json on 1 try
2019-05-22 18:14:25,843 INFO sleeping 5
```

**Comment by edsu** [comment]
But does the [filter endpoint](https://developer.twitter.com/en/docs/tweets/filter-realtime/api-reference/post-statuses-filter.html) allow you to limit by language? I'm not seeing it. You could create your own filter function using twarc as a library:

```python
import twarc

t = twarc.Twarc()

def tweets_by_lang(track, lang):
    for tweets in t.filter(track=track):
        if tweet['lang'] == lang:
            yield tweet

for tweet in tweets_by_lang('obama', 'fr'):
    print(tweet['full_text'])
```

Or you could post-process your data to limit accordingly. But I don't see a way to tell Twitter to limit the filter stream by language on their side, unless I'm missing something?

**Comment by tugrulz** [comment]
It does allow (https://developer.twitter.com/en/docs/tweets/filter-realtime/guides/basic-stream-parameters)

If you do postfiltering you will ask more data than necessary from Twitter which may result in getting rate limited.

**Comment by edsu** [comment]
Yes, good point about the rate limiting. I see language documented there -- that should be doable.

**Comment by edsu** [comment]
I've just released v1.7.1 which allows you to specify a language to filter by with the --lang option, which can repeat. For example to collect tweets that mention paris or madrid that are in Spanish or French:

    twarc filter paris,madrid --lang es --lang fr
